src/advanced_inv.cpp

```diff
@@ -1985,7 +1985,8 @@ bool advanced_inventory::move_content( item &src_container, item &dest_container
     }
 
     std::string err;
-    if( !dest_container.fill_with( src, err ) ) {
+    // @todo Allow buckets here, but require them to be on the ground or wielded
+    if( !dest_container.fill_with( src, err, false ) ) {
```

Comment:
Late to the party, but anyway:

Either an enum, a local constant (`const bool allow_bucket = false;` and use that as argument) or a separate, aptly named function (that does not have the parameter and simply forwards to the actual function).

Anyway, code is fine as it is.